store/postgres/src/detail.rs

```diff
@@ -277,25 +273,57 @@ pub(crate) fn deployment_statuses(
     use subgraph_deployment as d;
     use subgraph_error as e;
 
-    // Empty deployments means 'all of them'
-    if sites.is_empty() {
-        d::table
-            .left_outer_join(e::table.on(d::fatal_error.eq(e::id.nullable())))
-            .load::<(DeploymentDetail, Option<ErrorDetail>)>(conn)?
-            .into_iter()
-            .map(|(detail, error)| status::Info::try_from(DetailAndError(detail, error, sites)))
-            .collect()
-    } else {
-        let ids: Vec<_> = sites.into_iter().map(|site| site.id).collect();
+    // First, we fetch all deployment information along with any fatal errors.
+    // Subsequently, we fetch non-fatal errors and we group them by deployment
+    // ID.
+
+    let details_with_fatal_error = {
+        let join = e::table.on(e::id
+            .nullable()
+            .eq(d::fatal_error)
+            .and(e::subgraph_id.eq(d::deployment)));
+
+        // Empty deployments means 'all of them'
+        if sites.is_empty() {
+            d::table
+                .left_outer_join(join)
+                .load::<(DeploymentDetail, Option<ErrorDetail>)>(conn)?
+        } else {
+            d::table
+                .left_outer_join(join)
+                .filter(d::id.eq_any(sites.iter().map(|site| site.id)))
+                .load::<(DeploymentDetail, Option<ErrorDetail>)>(conn)?
+        }
+    };
 
-        d::table
-            .left_outer_join(e::table.on(d::fatal_error.eq(e::id.nullable())))
-            .filter(d::id.eq_any(&ids))
-            .load::<(DeploymentDetail, Option<ErrorDetail>)>(conn)?
-            .into_iter()
-            .map(|(detail, error)| status::Info::try_from(DetailAndError(detail, error, sites)))
-            .collect()
-    }
+    let mut non_fatal_errors = {
+        let join = e::table.on(e::id
+            .eq(dsl::any(d::non_fatal_errors))
+            .and(e::subgraph_id.eq(d::deployment)));
+
+        if sites.is_empty() {
+            d::table
+                .inner_join(join)
+                .select((d::id, e::all_columns))
+                .load::<(DeploymentId, ErrorDetail)>(conn)?
+        } else {
+            d::table
+                .inner_join(join)
+                .filter(d::id.eq_any(sites.iter().map(|site| site.id)))
+                .select((d::id, e::all_columns))
+                .load::<(DeploymentId, ErrorDetail)>(conn)?
+        }
+        .into_iter()
+        .into_group_map()
+    };
+
+    details_with_fatal_error
+        .into_iter()
+        .map(|(detail, fatal)| {
+            let non_fatal = non_fatal_errors.remove(&detail.id).unwrap_or(vec![]);
```

Comment:
What's being removed here?

Reply:
Line [299](https://github.com/graphprotocol/graph-node/pull/3350/files/29e4cb5ba6ae2852e2a9a8ca5c798425a352bac5#diff-52909c02a20cf838fc6f0a0b5e9b34c575abe78fbe463edd3022a43b12ae233dR299) collects non-fatal errors in a `HashSet` by `DeploymentId`. As we go over `DeploymentDetail`'s, we're taking ownership of its non-fatal errors by removing them from the `HashSet`. Then we build `status::Info` at line [324](https://github.com/graphprotocol/graph-node/pull/3350/files/29e4cb5ba6ae2852e2a9a8ca5c798425a352bac5#diff-52909c02a20cf838fc6f0a0b5e9b34c575abe78fbe463edd3022a43b12ae233dR324).

Reply:
Got it thanks for clarifying.